S3/Config.py

```diff
@@ -275,38 +281,46 @@ def role_refresh(self):
             except:
                 warning("Could not refresh role")
 
-    def env_config(self):
-        cred_content = ""
+    def aws_credential_file(self):
         try:
-            cred_file = open(os.environ['AWS_CREDENTIAL_FILE'],'r')
-            cred_content = cred_file.read()
+            aws_credential_file = os.path.expanduser('~/.aws/credentials') 
+            if 'AWS_CREDENTIAL_FILE' in os.environ and os.path.isfile(os.environ['AWS_CREDENTIAL_FILE']):
+                aws_credential_file = os.environ['AWS_CREDENTIAL_FILE']
+
+            config = PyConfigParser()
+
+            debug("Reading AWS credentials from", aws_credential_file)
+            config.read(aws_credential_file)
+
+            profile = "default"
+            if 'AWS_PROFILE' in os.environ:
+                profile = os.environ['AWS_PROFILE']
+            
+            debug("Using AWS profile '{}'".format(profile))
+
+            # trying to read aws_access_key_id from credentials file
+            profile_access_key = config.get(profile, 'aws_access_key_id')
+            debug('Setting "aws_access_key_id" from file {} as "access_key"'.format(aws_credential_file))
+            Config().update_option('access_key', config_unicodise(profile_access_key))
+
+            # trying to read aws_secret_access_key from credentials file
+            profile_secret_key = config.get(profile, 'aws_secret_access_key')
+            debug('Setting "aws_secret_access_key" from file {} as "secret_key"'.format(aws_credential_file))
+            Config().update_option('secret_key', config_unicodise(profile_secret_key))
+            
```

Comment:
Based on the previous code behavior, I think that you should take care that maybe only a part of access/secret/token keys could be stored in the config file.
Like only the session token, or only the access_key, ...
The rest being provided on the command line for example.